src/stdlib_hashmap_chaining.f90

```diff
@@ -0,0 +1,848 @@
+!!
+!! Returns the other data associated with the inverse table index
+!! Arguments:
+!!     map    - a chaining hash map
+!!     key    - the key associated with a map entry
+!!     other  - the other data associated with the key
+!!     exists - a logical flag indicating whether an entry with that key exists
+!
+        class(chaining_hashmap_type), intent(inout) :: map
+        type(key_type), intent(in)                  :: key
+        type(other_type), intent(out)               :: other
+        logical, intent(out), optional              :: exists
+
+        integer(int_index) :: inmap
+        character(*), parameter :: procedure = 'GET_OTHER_DATA'
+
+        call in_chain_map(map, inmap, key)
+        if ( inmap <= 0 .or. &
+             inmap > size(map % inverse, kind=int_index ) ) then
+            if ( present(exists) ) then
+                exists = .false.
+                return
+            else
+                error stop submodule_name // ' % ' // procedure // ': ' // &
+                    invalid_inmap
+            end if
+        else if ( associated( map % inverse(inmap) % target ) ) then
+            if (present(exists) ) exists = .true.
+            call copy_other( map % inverse(inmap) % target % other, other )
+        else
+            if ( present(exists) ) then
+                exists = .false.
+                return
+            else
+                error stop submodule_name // ' % ' // procedure // ': ' // &
+                    map_consist_fault
+            end if
+        end if
+
+    end subroutine get_other_chaining_data
+
+
+    subroutine in_chain_map(map, inmap, key)
+!! Version: Experimental
+!!
+!! Returns the index into the INVERSE array associated with the KEY
+!! Arguments:
+!!     map   - the hash map of interest
+!!     inmap - the returned index into the INVERSE array of entry pointers
+!!     key   - the key identifying the entry of interest
+!
+        class(chaining_hashmap_type), intent(inout) :: map
+        integer(int_index), intent(out)             :: inmap
+        type(key_type), intent(in)                  :: key
+
+        integer(int_hash)                      :: hash_val, hash_index
+        type(chaining_map_entry_type), pointer :: gentry, pentry, sentry
+
+        if ( map % probe_count > inmap_probe_factor * map % call_count ) then
+            if ( map % nbits < max_bits .AND. &
+                 map % num_entries > size( map % slots, kind=int_index ) ) then
+                call expand_slots(map)
+            end if
+        end if
+        map % call_count = map % call_count + 1
+        hash_val = map % hasher( key )
+        hash_index = fibonacci_hash( hash_val, map % nbits )
+        pentry => map % slots(hash_index) % target
+        sentry => pentry
+
+        climb_chain: do
+            gentry => pentry
+            map % probe_count = map % probe_count + 1
+            if (.not. associated( gentry ) ) then
+                write(error_unit,*) "gentry not associated"
```

Comment:
This line feels odd, as far as I can't tell we didn't find the index and we are returning `inmap=0` but also it writes to `error_unit`, plus this subroutine is only used internally, right?

You should probably mention that this is how you are signaling a missing index in its docstring and remove this line.

Reply:
I believe I have now addressed this issue.